src/app/app.component.ts

```diff
@@ -1,10 +1,35 @@
 import { Component } from '@angular/core';
+import { FormBuilder, FormGroup, Validators } from '@angular/forms';
 
 @Component({
   selector: 'app-root',
   templateUrl: './app.component.html',
   styleUrls: ['./app.component.scss']
 })
 export class AppComponent {
-  title = 'wolox-books';
+  rForm: FormGroup;
+  post: any;
+  firstName: string = '';
+  lastName: string = '';
+  email: string = '';
+  passwordConfirmation: string = '';
+  messageAlert: string = 'Campo obligatorio'
+
+  constructor(private fb: FormBuilder){
+    this.rForm = fb.group({
```

Comment:
No need to declare the form in the constructor, you can create it as a class attribute